src/main/scala/mesosphere/mesos/Constraints.scala

```diff
@@ -125,4 +126,85 @@ object Constraints {
 
   def meetsConstraint(tasks: Iterable[MarathonTask], offer: Offer, constraint: Constraint): Boolean =
     new ConstraintsChecker(tasks, offer, constraint).isMatch
+
+  /**
+    * Select tasks to kill while maintaining the constraints of the application definition.
+    * Note: It is possible, that the result of this operation does not select as much tasks as needed.
+    * @param app the application definition of the tasks.
+    * @param runningTasks the list of running tasks to filter
+    * @param toKillCount the expected number of tasks to select for kill
+    * @return the selected tasks to kill. The number of tasks will not exceed toKill but can be less.
+    */
+  //scalastyle:off return
+  def selectTasksToKill(app: AppDefinition, runningTasks: Set[MarathonTask], toKillCount: Int): Set[MarathonTask] = {
+    require(toKillCount <= runningTasks.size, "Can not kill more instances than running")
+
+    //short circuit, if all tasks shall be killed
+    if (runningTasks.size == toKillCount) return runningTasks
+
+    //currently, only the GROUP_BY operator is able to select tasks to kill
+    val distributions = app.constraints.filter(_.getOperator == Operator.GROUP_BY).map { constraint =>
+      def groupFn(task: MarathonTask): Option[String] = constraint.getField match {
+        case "hostname"    => Some(task.getHost)
+        case field: String => task.getAttributesList.asScala.find(_.getName == field).map(_.getText.getValue)
+      }
+      GroupByDistribution(constraint, runningTasks.groupBy(groupFn).values.toSeq)
+    }
+
+    //short circuit, if there are no constraints to align with
+    if (distributions.isEmpty) return Set.empty
+
+    var toKillTasks = Set.empty[MarathonTask]
+    var flag = true
+    while (flag && toKillTasks.size != toKillCount) {
+      val tried =
+        distributions.toSeq.sortBy(_.distributionDifference(toKillTasks)).reverseIterator //highest diff first
+          .flatMap(_.selectTaskToKill(toKillTasks)) ++ //select all tasks to kill
```

Comment:
What about changing the comment to something a bit more descriptive like:

For each distribution, select the tasks of the "largest" value group. The tasks in "toKillTasks" will not be counted.